test/remote/gateways/remote_cyber_source_rest_test.rb

```diff
@@ -63,13 +63,38 @@ def test_failure_authorize_with_declined_credit_card
 
   def test_successful_purchase
     response = @gateway.purchase(@amount, @visa_card, @options)
-
     assert_success response
     assert response.test?
     assert_equal 'AUTHORIZED', response.message
     assert_nil response.params['_links']['capture']
   end
 
+  def test_successful_void_a_payment
+    purchase = @gateway.purchase(@amount, @visa_card, @options)
+    response = @gateway.void(purchase.authorization, @options)
+    assert response.params['id'].present?
+    assert_equal 'VOIDED', response.message
+    assert_nil response.params['_links']['capture']
+  end
+
+  def test_successful_void_a_capture
+    @options[:void_type] = 'captures'
+    purchase = @gateway.purchase(@amount, @visa_card, @options)
+    response = @gateway.void(purchase.authorization, @options)
+    assert_success response
+    assert response.params['id'].present?
+    assert_equal 'VOIDED', response.message
+    assert_nil response.params['_links']['capture']
+  end
+
+  def test_successful_verify
+    response = @gateway.verify(@visa_card, @options)
+    assert_success response
+    assert response.params['id'].present?
+    assert_equal 'AUTHORIZED', response.message
+    assert_nil response.params['_links']['capture']
+  end
```

Comment:
💬 Comment:

please check the documentation to see if you can build some scenarios with failing transactions.

Reply:
Done!